kraken-build/tests/kraken_std/integration/cargo/test_cargo_private_registry.py

```diff
@@ -169,24 +215,26 @@ def publish_lib_and_build_app(repository: CargoRepositoryWithAuth, tempdir: Path
 #       API endpoint. Until we can find a way to change this behavior, we need the container port and the host port
 #       to be the same, which prevents us from running multiple test-scoped fixtures in parallel.
 @pytest.fixture(scope="session")
-def private_registry(docker_service_manager: DockerServiceManager) -> str:
-    container = docker_service_manager.run(
-        "ghcr.io/d-e-s-o/cargo-http-registry:latest",
-        [
-            "/tmp/test-registry",
-            "--addr",
-            "0.0.0.0:35504",
-        ],
-        ports=["35504:35504"],
-        detach=True,
-    )
+def private_registry(docker_service_manager: DockerServiceManager) -> Iterator[str]:
+    with file_lock(
+        Path("/tmp/kraken_cargo_private_registry_lock")
+    ):  # We hardcode a port, only a single instance must exist at the same time
+        container = docker_service_manager.run(
+            "ghcr.io/d-e-s-o/cargo-http-registry:sha-2edffd8",  # TODO(Tpt): hardcoded because latest docker images are broken
```

Comment:
Example of CI from develop that fails because of that: https://github.com/kraken-build/kraken/actions/runs/13418579166/job/37803567596